### PR TITLE
feat: supported streaming emulation

### LINKED ISSUE #156: Support o1-mini and o1-preview models
The main limitation of the models is the missing streaming feature.
Thus in the adapter we need to emulate the streaming otherwise the DIAL Chat won't work:

```
NON_STREAMING_DEPLOYMENTS=o1-mini-debug,o1-preview-debug
```

### PR DESCRIPTION
**Pull request by adubovik**
Resolves #156

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-157.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-openai-pr-157.nightly-test.deltixhub.io
>E2E tests status: success

File: aidial_adapter_openai/app.py [hunk 1]
```diff
@@ -212,13 +197,48 @@ async def embedding(deployment_id: str, request: Request):
...
+
+    if isinstance(e, APIError):
+        raise DialException(
+            status_code=getattr(e, "status_code", None) or 500,
+            message=e.message,
+            type=e.type,
+            code=e.code,
+            param=e.param,
+            display_message=None,
+        )
```
Comment (roman-romanov-o) [inline aidial_adapter_openai/app.py:237]: Minor:

maybe let's use match/case for that?

```python
match e:
        case APIStatusError():
            r = e.response
            return Response(
                content=r.content,
                status_code=r.status_code,
                headers=r.headers,
            )
        
        case APITimeoutError():
            raise DialException(
                status_code=504,
                type="timeout",
                message="Request timed out",
                display_message="Request timed out. Please try again later.",
            )
        
        case APIConnectionError():
            raise DialException(
                status_code=502,
                type="connection",
                message="Error communicating with OpenAI",
                display_message="OpenAI server is not responsive. Please try again later.",
            )
        
        case APIError():
            raise DialException(
                status_code=getattr(e, "status_code", None) or 500,
                message=e.message,
                type=e.type,
                code=e.code,
                param=e.param,
                display_message=None,
            )
        
        case _:
            raise DialException(
                status_code=500, 
                message="Unknown OpenAI exception",
                # =================
                # Or maybe without "openai", just "unknown_upstream_exception" 
                code="openai_unknown_exception"
            )
```

And wrap unknown exception into some `code` so we could track such errors?

**Review by roman-romanov-o — APPROVED** [review]

File: aidial_adapter_openai/app.py (hunk 1 above)
Comment (roman-romanov-o) [reply aidial_adapter_openai/app.py:237]: But I guess it's not necessary for this PR, since we had same logic before, without handling unknown exceptions

Comment (adubovik) [reply aidial_adapter_openai/app.py:237]: Yes, let's address it in another PR.

DIAL SDK handles all kinds of errors gracefully, but the adaptor isn't based on the SDK, so we have to do it manually :(

**Pull request merged**